### DISCUSSION TITLE
Get from Expression Left, Operator and Right and Save into list

### DISCUSSION THREAD

**Discussion by rider87** [JSQLParser/JSqlParser#1262] [General]
Hi,

I saw I can get from a Expression the left, operator and right side...
I would like to extract that into a List<MyList> to do something else later on...

What I´m trying to do is to get the whole Expression:
- Convert this expression into Java Code
- Validate this against Java...
- E.g. you have Expression 1==1... I would like to check this directly in Java if(1==1) e.g

`	private static String test(BinaryExpression expr) {

		boolean test; 

		ExpressionVisitorAdapter expressionVisitorAdapter = new ExpressionVisitorAdapter();

		expr.accept(new ExpressionVisitorAdapter() {
			protected void visitBinaryExpression(BinaryExpression expr) {
				if (expr instanceof ComparisonOperator) {

					Object left = expr.getLeftExpression();
					Object operator = expr.getLeftExpression();
					Object right = expr.getLeftExpression();

					System.out.println("left=" + expr.getLeftExpression() + "  op=" + expr.getStringExpression()
							+ "  right=" + expr.getRightExpression());
				}
							
				super.visitBinaryExpression(expr);
			}
		});
		
		return expr.getLeftExpression().toString();
	}`

Is there a possibility to get this? Due tot he fact it´s procted I cannot add a List<> from outside the :
**expr.accept(new ExpressionVisitorAdapter() {**

Any ideas?

**Comment by manticore-projects** [comment]
Greetings.

You can define a `final` list outside of the `accept` and add elements to this list inside the `visit` method.
However, I struggle to understand what you would like to achieve as we are not talking about a list but rather about a tree of nested expressions.

**Comment by manticore-projects** [comment]
```java
private static String test(BinaryExpression expr) {

    boolean test;
    final ArrayList<Expression> expressions = new ArrayList<>();
    
    ExpressionVisitorAdapter expressionVisitorAdapter = new ExpressionVisitorAdapter();

    expr.accept(new ExpressionVisitorAdapter() {
      protected void visitBinaryExpression(BinaryExpression expr) {
        if (expr instanceof ComparisonOperator) {
            ComparisonOperator comparisonOperator = (ComparisonOperator) expr;

         Expression leftExpression = comparisonOperator.getLeftExpression();
         String operator = comparisonOperator.getStringExpression();
         Expression rightExpression =  comparisonOperator.getRightExpression();

         System.out.println("left=" + leftExpression + "  op="
              + operator + "  right=" + rightExpression);
         
         expressions.add(leftExpression);
         expressions.add(rightExpression);
        }

        super.visitBinaryExpression(expr);
      }
    });
    
    StringBuilder builder = new StringBuilder();
    for (Expression expression:expressions)
        builder.append(expression.toString());
    
    return builder.toString();
  }
```
